lib/server.js

```diff
@@ -79,14 +78,148 @@ function createServer(picker, log) {
+                    err = new errors.BadRequestError({
+                        statusCode: 400
+                    }, 'Invalid value for limit parameter: ' + limit);
+                    return (next(err));
+                }
+                break;
+            case 'after_id':
+                /* XXX add code to sanity check id format */
+                after_id = req.params.after_id;
+                break;
+            case 'only_id':
+                /* XXX add code to sanity check id format */
+                only_id = req.params.only_id;
+                break;
+            default:
+                err = new errors.BadRequestError({
+                    statusCode: 400
+                    }, 'Invalid parameter: ' + param);
+                return (next(err));
+        }
+        if (after_id !== null && only_id !== null) {
+            err = new errors.BadRequestError({
+                statusCode: 400
+                }, 'after_id and only_id params are mutually exclusive');
+            return (next(err));
+        }
+    }
+
+    if (only_id !== null) {
+        /* JSSTYLED */
+        idx = sharkMap.findIndex(obj => obj.manta_storage_id === only_id);
+
+        if (idx === -1) {
+            return (next(new errors.NotFoundError()));
+        }
+        res.send(200, sharkMap[idx]);
+        return (next());
+    }
+
+    /*
+     * The fast path.  If after_id was not specified AND the limit is greater
+     * than or equal to the total number of results, then just return the
+     * entire shark map.
+     */
+    if (after_id === null && limit >= nSharks) {
+        res.send(200, sharkMap);
+        return (next());
+    }
+
+    if (after_id !== null) {
+        var last_idx =
+            /* JSSTYLED */
+           sharkMap.findIndex(obj => obj.manta_storage_id === after_id);
+
+        if (last_idx === -1) {
+            /*
+             * The shark referenced by after_id isn't in the map - possibly
+             * because it was removed.  We need to figure out where to set the
+             * cursor now.  Basically we want to find the index N in the
+             * sharkMap array where:
+             *
+             * Map[N-1].manta_storage_id < after_id < Map[N].manta_storage_id
+             */
+            var still_looking = true;
+
+            /*
+             * Before we search, let's check the obvious edge cases:
+             */
+            if (sharkMap[0].manta_storage_id > after_id) {
+                start_idx = 0;
+                still_looking = false;
+            } else if (sharkMap[nSharks - 1].manta_storage_id < after_id) {
+                return (next(new errors.NotFoundError()));
+            }
+
+            /*
+             * Ok, do a binary search through the array to find index N.
+             */
+            idx = Math.floor(nSharks / 2);
+            while (still_looking) {
+                console.log('setting idx to: ' + idx);
```

Comment:
We should try to use the bunyan logger.  It might be in the server object, or in the req.picker object.  If not we should consider adding it to the req object in the `_setPicker()` closure on line 75.

Reply:
Yeah - that was just debug code that neglected to remove.   I'll remove that line.